### PR TITLE
refactor(db/schema): store documentation of fields alongside code

### PR DESCRIPTION
**Pull request by jschmid1**
### Summary

Supersedes #10706 
Supersedes https://github.com/Kong/kong/pull/10150

This PR is designed to streamline our documentation process for all configurable fields within Kong, by positioning the documentation closer to their respective definitions. This strategic placement facilitates automatic generation of updated documentation, eliminating the need for multiple location updates.

At this point, I didn't mark the description field `required = true`, in order to prevent potential conflicts when transitioning to EE. Instead, we should approach this as an iterative process, only making description fields mandatory once we are confident of their completeness.

Co-authored-by: Angel Guarisma angel.guarisma@konghq.com

### Checklist

~- [ ] The Pull Request has tests~
~- [ ] There's an entry in the CHANGELOG~
~- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE~

### Full changelog

* adding `description` fields to all configurable fields.

### Issue reference

https://konghq.atlassian.net/browse/KAG-475

### CODE REVIEW

File: kong/db/schema/metaschema.lua
Comment (hbagdi) [inline kong/db/schema/metaschema.lua:75]: Let's add a minimum and a maximum length limit to these.

Comment (jschmid1) [reply kong/db/schema/metaschema.lua]: Added a minimum length of `10` and a maximum length of `500` characters. This tolerates the _existing_ descriptions we have in our docs (with one needed change). The longest description we have is 447 characters long.

As a reference here is an example text with 500 characters:

```
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla efficitur urna neque, eu vehicula libero tristique ac. Vivamus lacinia turpis vel mi feugiat, eu pharetra dui facilisis. Sed in sapien id odio bibendum fermentum nec ut justo. Aenean imperdiet lacinia nisi, nec tempus odio euismod ut. Pellentesque at magna ac ligula commodo aliquet eget a arcu. Proin vitae dolor vel mi vulputate condimentum. Suspendisse bibendum purus velit, ut condimentum odio finibus nec. Fusce scelerisque, quam ut sollicitudin fringilla, sapien mauris tristique odio, vel dictum lectus libero id ante. Integer ullamcorper malesuada sapien, non gravida sapien lobortis vel. Etiam euismod, lorem nec cursus hendrerit, lectus nulla pharetra augue, id pulvinar dui lacus eu turpis. In hac habitasse platea dictumst. Ut sit amet tortor nunc.
```

And an example text with 10 characters.

```
Lorem ipsum d
```

Comment (hbagdi) [reply kong/db/schema/metaschema.lua]: Looks good.

**Comment by hbagdi** [comment]
@jschmid1 Please rebase.

**Comment by hbagdi** [comment]
@chronolaw Can you please fix the failing test?

**Comment by Guaris** [comment]
@hbagdi @kikito @jschmid1 Can we merge this?

File: kong/plugins/response-ratelimiting/schema.lua
Comment (kikito) [inline kong/plugins/response-ratelimiting/schema.lua:51]:
```suggestion
    "The rate-limiting policies to use for retrieving and incrementing the limits.",
```

File: kong/plugins/statsd/schema.lua
Comment (kikito) [inline kong/plugins/statsd/schema.lua:191]: The style on the field descriptions on this plugin is a bit inconsistent:
* Using "The xxx that does yyy" formula: "The IP address ..."
* Just saying "Does yyy": "Determines what sort of event... "
* Just saying "xxx": "List of status code ..."

It seems like the first style ("The xxx that does yyy") is the most usual one. I propose using that style in all the fields.

File: kong/tools/queue_schema.lua
Comment (kikito) [inline kong/tools/queue_schema.lua:11]: I think these descriptions should start in Uppercase and end in a dot.

```suggestion
      description = "Maximum number of entries that can be processed at a time."
```

**Review by kikito — CHANGES_REQUESTED** [review]
I think this PR is mostly fine, my only real complain is with the "FILL ME WITH SOMETHING MEANINGFUL" descriptions, we should not merge this with those in, unless there's a valid reason

File: kong/db/schema/entities/clustering_data_planes.lua
Comment (Guaris) [inline kong/db/schema/entities/clustering_data_planes.lua:32]:
```suggestion
                      description = "The status of the clustering data planes sync.",
```

File: kong/db/schema/entities/consumers.lua
Comment (Guaris) [inline kong/db/schema/entities/consumers.lua:24]:
```suggestion
        description = "Stores the existing unique ID of the consumer.",
```

File: kong/db/schema/entities/parameters.lua
Comment (Guaris) [inline kong/db/schema/entities/parameters.lua:12]:
```suggestion
    { key            = { description = "They key value of a parameter.", type = "string", required = true, unique = true, }, },
```

File: kong/db/schema/entities/parameters.lua
Comment (Guaris) [inline kong/db/schema/entities/parameters.lua:13]:
```suggestion
    { value          = { description = "The value attached to the key.", type = "string", required = true, }, },
```

File: kong/db/schema/entities/plugins.lua
Comment (Guaris) [inline kong/db/schema/entities/plugins.lua:24]:
```suggestion
    { consumer = { description = "If set, the plugin will activate only for requests where the specified has been authenticated.", type = "foreign", reference = "consumers", default = null, on_delete = "cascade", }, },
```

File: kong/db/schema/entities/plugins.lua
Comment (Guaris) [inline kong/db/schema/entities/plugins.lua:25]:
```suggestion
    { config = { description = "The configuration properties for the Plugin.", type = "record", abstract = true, }, },
```

File: kong/db/schema/entities/routes.lua
Comment (Guaris) [inline kong/db/schema/entities/routes.lua:52]:
```suggestion
      { preserve_host  = { description = "When matching a Route via one of the hosts domain names, use the request Host header in the upstream request headers.", type = "boolean", required = true, default = false }, },
```

File: kong/db/schema/entities/routes.lua
Comment (Guaris) [inline kong/db/schema/entities/routes.lua:53]:
```suggestion
      { request_buffering  = { description = "Whether to enable request body buffering or not. With HTTP 1.1.", type = "boolean", required = true, default = true }, },
```

File: kong/db/schema/entities/routes.lua
Comment (Guaris) [inline kong/db/schema/entities/routes.lua:57]:
```suggestion
      { expression = { description = " The router expression.", type = "string", required = true }, },
```

File: kong/db/schema/entities/routes.lua
Comment (Guaris) [inline kong/db/schema/entities/routes.lua:125]:
```suggestion
      { request_buffering  = { description = "Whether to enable request body buffering or not. With HTTP 1.1.", type = "boolean", required = true, default = true }, },
```

File: kong/db/schema/entities/routes.lua
Comment (Guaris) [inline kong/db/schema/entities/routes.lua:126]:
```suggestion
      { response_buffering  = { description = "Whether to enable response body buffering or not.", type = "boolean", required = true, default = true }, },
```

File: kong/db/schema/entities/services.lua
Comment (Guaris) [inline kong/db/schema/entities/services.lua:44]:
```suggestion
    { tls_verify         = { description = "Whether to enable verification of upstream server TLS certificate.", type = "boolean", }, },
```

File: kong/db/schema/entities/services.lua
Comment (Guaris) [inline kong/db/schema/entities/services.lua:45]:
```suggestion
    { tls_verify_depth   = { description = "Maximum depth of chain while verifying Upstream server's TLS certificate.", type = "integer", default = null, between = { 0, 64 }, }, },
```

File: kong/db/schema/entities/snis.lua
Comment (Guaris) [inline kong/db/schema/entities/snis.lua:17]:
```suggestion
    { certificate  = { description = "The id (a UUID) of the certificate with which to associate the SNI hostname.", type = "foreign", reference = "certificates", required = true }, },
```

File: kong/db/schema/entities/tags.lua
Comment (Guaris) [inline kong/db/schema/entities/tags.lua:12]:
```suggestion
    { entity_name  = { description = "The name of the Kong Gateway entity being tagged.", type = "string", required = true }, },
```

File: kong/db/schema/entities/targets.lua
Comment (Guaris) [inline kong/db/schema/entities/targets.lua:31]:
```suggestion
    { target     = { description = "The target address (ip or hostname) and port.", type = "string", required = true, custom_validator = validate_target, }, },
```

File: kong/db/schema/entities/upstreams.lua
Comment (Guaris) [inline kong/db/schema/entities/upstreams.lua:192]:
```suggestion
    { hash_on_cookie = { description = "The cookie name to take the value from as hash input.", type = "string",  custom_validator = utils.validate_cookie_name }, },
```

File: kong/db/schema/entities/upstreams.lua
Comment (Guaris) [inline kong/db/schema/entities/upstreams.lua:198]:
```suggestion
    { slots = { description = "The number of slots in the load balancer algorithm.", type = "integer", default = 10000, between = { 10, 2^16 }, }, },
```

File: kong/db/schema/entities/upstreams.lua
Comment (Guaris) [inline kong/db/schema/entities/upstreams.lua:199]:
```suggestion
    { healthchecks = { description = "The array of healthchecks.", type = "record",
```

File: kong/db/schema/entities/upstreams.lua
Comment (Guaris) [inline kong/db/schema/entities/upstreams.lua:206]:
```suggestion
    { use_srv_name =  { description = "If set, the balancer will use SRV hostname.", type = "boolean", default = false, }, },
```

File: kong/db/schema/entities/vaults.lua
Comment (Guaris) [inline kong/db/schema/entities/vaults.lua:60]:
```suggestion
    { name = { description = "The name of the Vault that's going to be added.", type = "string", required = true }},
```

File: kong/db/schema/typedefs.lua
Comment (Guaris) [inline kong/db/schema/typedefs.lua:420]:
```suggestion
  description = "When using the `redis` policy with `redis_ssl` set to `true`, this property specifies the server name for the TLS extension Server Name Indication (SNI)."
```

File: kong/plugins/acme/schema.lua
Comment (Guaris) [inline kong/plugins/acme/schema.lua:58]:
```suggestion
  { https = { type = "boolean", default = false, description = "Boolean representation of https."}, },
```

File: kong/plugins/acme/schema.lua
Comment (Guaris) [inline kong/plugins/acme/schema.lua:61]:
```suggestion
  { kv_path = { type = "string", description = "KV prefix path."}, },
```

File: kong/plugins/acme/schema.lua
Comment (Guaris) [inline kong/plugins/acme/schema.lua:62]:
```suggestion
  { timeout = { type = "number", description = "Timeout in milliseconds."}, },
```

File: kong/plugins/acme/schema.lua
Comment (Guaris) [inline kong/plugins/acme/schema.lua:63]:
```suggestion
  { token = { type = "string", referenceable = true, description = "Consul ACL token."}, },
```

File: kong/plugins/acme/schema.lua
Comment (Guaris) [inline kong/plugins/acme/schema.lua:67]:
```suggestion
  { https = { type = "boolean", default = false, description = "Boolean representation of https." }, },
```

File: kong/plugins/acme/schema.lua
Comment (Guaris) [inline kong/plugins/acme/schema.lua:70]:
```suggestion
  { kv_path = { type = "string", description = "KV prefix path." }, },
```

File: kong/plugins/acme/schema.lua
Comment (Guaris) [inline kong/plugins/acme/schema.lua:71]:
```suggestion
  { timeout = { type = "number", description = "Timeout in milliseconds."}, },
```

File: kong/plugins/acme/schema.lua
Comment (Guaris) [inline kong/plugins/acme/schema.lua:72]:
```suggestion
  { token = { type = "string", referenceable = true, description = "Consul ACL token." }, },
```

File: kong/plugins/acme/schema.lua
Comment (Guaris) [inline kong/plugins/acme/schema.lua:73]:
```suggestion
  { tls_verify = { type = "boolean", default = true, description = "Turn on TLS verification." }, },
```

File: kong/plugins/acme/schema.lua
Comment (Guaris) [inline kong/plugins/acme/schema.lua:74]:
```suggestion
  { tls_server_name = { type = "string", description = "SNI used in request, default to host if omitted."  }, },
```

File: kong/plugins/acme/schema.lua
Comment (Guaris) [inline kong/plugins/acme/schema.lua:75]:
```suggestion
  { auth_method = { type = "string", default = "token", one_of = { "token", "kubernetes" }, description = "Auth Method, default to token, can be "token" or "kubernetes." } },
```

File: kong/plugins/acme/schema.lua
Comment (Guaris) [inline kong/plugins/acme/schema.lua:76]:
```suggestion
  { auth_path = { type = "string", description = "Vault's authentication path to use." }, },
```

File: kong/plugins/acme/schema.lua
Comment (Guaris) [inline kong/plugins/acme/schema.lua:77]:
```suggestion
  { auth_role = { type = "string", description = "The role to try and assign." }, },
```

File: kong/plugins/acme/schema.lua
Comment (Guaris) [inline kong/plugins/acme/schema.lua:78]:
```suggestion
  { jwt_path = { type = "string", description = "The path to the JWT." }, },
```

File: kong/plugins/acme/schema.lua
Comment (Guaris) [inline kong/plugins/acme/schema.lua:82]:
```suggestion
  { key_id = { type = "string", required = true, description = "The Key ID." } },
```

File: kong/plugins/acme/schema.lua
Comment (Guaris) [inline kong/plugins/acme/schema.lua:83]:
```suggestion
  { key_set = { type = "string", description = "The ID of the key set to associate the Key ID with." } }
```

File: kong/plugins/acme/schema.lua
Comment (Guaris) [reply kong/plugins/acme/schema.lua:58]: I have just added to fix this.

File: kong/plugins/statsd/schema.lua
Comment (jschmid1) [reply kong/plugins/statsd/schema.lua:191]: @Guaris Your call.

File: kong/tools/queue_schema.lua
Comment (kikito) [inline kong/tools/queue_schema.lua:17]:
```suggestion
      description = "Maximum number of (fractional) seconds to elapse after the first entry was queued before the queue starts calling the handler.",
```

File: kong/tools/queue_schema.lua
Comment (kikito) [inline kong/tools/queue_schema.lua:25]:
```suggestion
      description = "Maximum number of entries that can be waiting on the queue.",
```

File: kong/tools/queue_schema.lua
Comment (kikito) [inline kong/tools/queue_schema.lua:30]:
```suggestion
      description = "Maximum number of bytes that can be waiting on a queue, requires string content.",
```

File: kong/tools/queue_schema.lua
Comment (kikito) [inline kong/tools/queue_schema.lua:35]:
```suggestion
      description = "Time in seconds before the queue gives up calling a failed handler for a batch.",
```

File: kong/tools/queue_schema.lua
Comment (kikito) [inline kong/tools/queue_schema.lua:43]:
```suggestion
        description = "Time in seconds before the initial retry is made for a failing batch."
```

File: kong/tools/queue_schema.lua
Comment (kikito) [inline kong/tools/queue_schema.lua:50]:
```suggestion
      description = "Maximum time in seconds between retries, caps exponential backoff."
```

**Review by kikito — APPROVED** [review]
Approving with suggestions on queue fields

**Review by kikito — APPROVED** [review]
Approving with suggestions on queue fields

**Comment by outsinre** [comment]
This PR may introduce bug to EE, considering the EE schemas usually have different field attributes. Need careful cherry-pick.

File: kong/db/schema/metaschema.lua
Comment (hbagdi) [reply kong/db/schema/metaschema.lua:75]: Temporary. See https://github.com/Kong/kong-ee/pull/5822

**Pull request merged**
